### ISSUE TITLE
Error: Newton's method failed to converge after 100 iterations

### ISSUE THREAD

**Issue by ElShapovalov** [goldwasserexchange/bond-calculator#63]
Hi! I have bond with next params: 
```
{ 
  maturity: 2031-08-15T00:00:00.000Z,
  redemption: 100,
  settlement: 2018-04-25T00:00:00.000Z,
  frequency: 2,
  rate: 0.09,
  convention: '30U/360' 
}
```
When i try to calculate yield with price 58.4, i get error "Newton's method failed to converge after 100 iterations"

But in EXCEL this bond works fine.

Help me please!

**Comment by jogold** [comment]
Hi @ElShapovalov,

I will increase max iterations to 200 and release and new version asap.

Thanks for reporting this!

**Comment by ElShapovalov** [comment]
I think it is wrong. Becaouse in description [YIELD](https://support.office.com/ru-ru/article/%D0%94%D0%9E%D0%A5%D0%9E%D0%94-%D1%84%D1%83%D0%BD%D0%BA%D1%86%D0%B8%D1%8F-%D0%94%D0%9E%D0%A5%D0%9E%D0%94-f5f5ca43-c4bd-434f-8bd2-ed3c9727a4fe?ui=ru-RU&rs=ru-RU&ad=RU) on russian language writen "**not more 100 iterations**", maybe need return value after 100 iterations.

And i found bug with other bond:
```
{ 
maturity: 2044-12-15T00:00:00.000Z,
redemption: 100,
settlement: 2018-04-28T00:00:00.000Z,
frequency: 4,
rate: 0.04721,
convention: '30U/360' 
}
```
`price: 50 `

module can not get previous coupone in **coupon.js** **dates** functions.
If set range from -3 then it works fine as EXCEL YILED function

**Comment by jogold** [comment]
Hi @ElShapovalov,

It should be fixed in v0.1.8.
